addons/account/data/payment_receipt_data.xml

```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <data noupdate="1">
+
+        <record id="account_payment_receipt_value" model="ir.values">
+            <field name="model_id" ref="account.model_account_payment" />
+            <field name="name">Print Receipt</field>
+            <field name="key2">client_action_multi</field>
+            <field name="value" eval="'ir.actions.report.xml,' +str(ref('action_report_payment_receipt'))" />
+            <field name="key">action</field>
+            <field name="model">account.payment</field>
+        </record>
+
+        <!--Email template -->
+        <record id="email_template_payment_receipt" model="mail.template">
+            <field name="name">Payment Receipt - Send by Email</field>
+            <field name="email_from">${(object.user_id.email and '%s &lt;%s&gt;' % (object.env.user.name, object.env.user.email) or '')|safe}</field>
+            <field name="subject">${object.company_id.name} Payment Receipt (Ref ${object.name or 'n/a' })</field>
+            <field name="partner_to">${object.partner_id.id}</field>
+            <field name="model_id" ref="account.model_account_payment"/>
+            <field name="auto_delete" eval="True"/>
+            <field name="report_template" ref="account.action_report_payment_receipt"/>
+            <field name="report_name">${(object.name or '').replace('/','-')}</field>
+            <field name="lang">${object.partner_id.lang}</field>
+            <field name="body_html"><![CDATA[
+                <p>
+                    Dear ${object.partner_id.name},
+                </p>
+                <p>Thank you for your payment.<br />
+                Here is your payment receipt <strong>${(object.name or '').replace('/','-')}</strong>
+                amounting to <strong>${object.amount} ${object.currency_id.name}</strong>
+                from ${object.company_id.name}.
+                </p>
+                <p>If you have any questions, please do not hesitate to contact us.</p>
+                <p>Best regards,</p>
```

Comment:
I think you should set user_signature to False and add object.user_id.signature directly in the email template. Indeed as this email is coming from "object.user_id" which may be different from the current user, we have to be sure the correct signature is added.